erpnext/projects/doctype/task/task.py

```diff
@@ -104,6 +106,33 @@ def update_depends_on(self):
 
 	def update_nsm_model(self):
 		frappe.utils.nestedset.update_nsm(self)
+	
+	def validate_default_project(self):
+		""" Validate projects table in task."""
+
+		unique_projects = list(set([project.project for project in self.projects]))
+		if len(self.projects) != len(unique_projects):
+			task_projects = list([project.project for project in self.projects])
```

Comment:
```suggestion
		_projects = [project.project for project in self.projects]
		if len(_projects) != len(list(set(_projects))):
			task_projects = _projects
```
- Pretty up this code, this will reduce your one for loop